packages/tailwindcss/src/compat/config/resolve-config.ts

```diff
@@ -0,0 +1,228 @@
+import { segment } from '../../utils/segment'
+import { deepMerge, isPlainObject } from './deep-merge'
+import {
+  type LoadedPlugin,
+  type LoadedPreset,
+  type RawPlugin,
+  type ResolvedConfig,
+  type ResolvedContentConfig,
+  type ResolvedThemeValue,
+  type ThemeValue,
+  type UserConfig,
+} from './types'
+
+interface ResolutionContext {
+  configs: UserConfig[]
+  plugins: RawPlugin[]
+  content: ResolvedContentConfig
+  theme: Record<string, ThemeValue>
+  extend: Record<string, ThemeValue[]>
+  result: ResolvedConfig
+}
+
+let minimal: ResolvedConfig = {
+  important: false,
+  content: {
+    files: [],
+    transform: {},
+  },
+  blocklist: [],
+  prefix: '',
+  separator: ':',
+  darkMode: 'media',
+  theme: {},
+  plugins: [],
+}
+
+export function resolveConfig(configs: UserConfig[]): ResolvedConfig {
```

Comment:
Should we simplify this for now to just be for resolving multiple themes rather than entire configs? Then use whatever we end up with here as part of resolving the whole config when we actually start working on support for JS config files? Feels wrong to me to add this stuff prematurely as part of this PR, even though of course it will eventually be needed when we actually do that work.

Reply:
Done.